addons/google_calendar/controllers/main.py

```diff
@@ -1,59 +1,58 @@
-import openerp.addons.web.http as http
-from openerp.addons.web.http import request
+# -*- coding: utf-8 -*-
+from openerp import http
+from openerp.http import request
 
 
-class google_calendar_controller(http.Controller):
+class GoogleCalendarController(http.Controller):
 
     @http.route('/google_calendar/sync_data', type='json', auth='user')
     def sync_data(self, arch, fields, model, **kw):
         """
-            This route/function is called when we want to synchronize openERP calendar with Google Calendar
+            This route/function is called when we want to synchronize Odoo calendar with Google Calendar
             Function return a dictionary with the status :  need_config_from_admin, need_auth, need_refresh, success if not calendar_event
-            The dictionary may contains an url, to allow OpenERP Client to redirect user on this URL for authorization for example
+            The dictionary may contains an url, to allow Odoo Client to redirect user on this URL for authorization for example
         """
 
         if model == 'calendar.event':
-            gs_obj = request.registry['google.service']
-            gc_obj = request.registry['google.calendar']
+            GoogleCalendar = request.env['google.calendar'].with_context(kw.get('local_context'))
 
             # Checking that admin have already configured Google API for google synchronization !
-            client_id = gs_obj.get_client_id(request.cr, request.uid, 'calendar', context=kw.get('local_context'))
+            client_id = request.env['google.service'].get_client_id('calendar')
```

Comment:
what will be context here ? no need to use with_context (update context for google service ) ?